packages/debug/index.js

```diff
@@ -9,4 +9,19 @@ module.exports = Object.assign({}, addonBaseConfig, {
   externalDependenciesForPrivateModule() {
     return [];
   },
+  isEnabled() {
+    const { hasOwnProperty } = Object.prototype;
+    const config = this.parent.config(this.app.env);
+    const userDefinedProperty = hasOwnProperty.call(config, 'includeDebug');
+    if (userDefinedProperty) {
+      // use the user defined property from environment.js
+      return config.includeDebug;
+    } else if (config.APP.name && config.APP.name === 'ember-data') {
```

Comment:
`config.App.name` I don't think is what we want to check here, what we want to check is if the parent addon is ember-data. Potentially we could have the `ember-data` package set a different default config variable and the debug package wouldn't need to know about the config switch. For this approach I think the `parent` property should have this info. cc @rwjblue for thoughts.